akka-cluster/src/main/scala/akka/cluster/ClusterActorRefProvider.scala

```diff
@@ -64,7 +61,31 @@ private[akka] class ClusterActorRefProvider(
    */
   override protected def createDeployer: ClusterDeployer = new ClusterDeployer(settings, dynamicAccess)
 
-  override protected def showDirectUseWarningIfRequired(): Unit = ()
+  override def actorOf(
+      system: ActorSystemImpl,
+      props: Props,
+      supervisor: InternalActorRef,
+      path: ActorPath,
+      systemService: Boolean,
+      deploy: Option[Deploy],
+      lookupDeploy: Boolean,
+      async: Boolean): InternalActorRef = {
+    val ref = super.actorOf(system, props, supervisor, path, systemService, deploy, lookupDeploy, async)
+    ref match {
+      case remoteRef: RemoteActorRef if !remoteSettings.UseUnsafeRemoteFeaturesWithoutCluster =>
+        val remoteAddress = remoteRef.path.address
+        if (Cluster(system).state.members.exists(_.address == remoteAddress))
+          remoteRef
+        else {
+          log.warning(
+            "Remote deploy of [{}] outside this cluster is not allowed, falling back to local.",
+            remoteAddress)
+          // `super.local` is private, this creates a LARP if systemService=true
+          super.actorOf(system, props, supervisor, path, systemService = true, deploy, lookupDeploy, async)
```

Comment:
I don't think it's correct to change `systemService`, at least the original meaning of that is something completely different (I think it has to do with `/user/` vs `/system/` actor paths. Can you open up `super.local`?

Reply:
I can if I change it from private

Reply:
It's currently `private val local = new LocalActorRefProvider` which I commented, thus the not ideal use of `systemService`. If we're ok with `private[akka] val local = new LocalActorRefProvider...` I can do that.